Algorithm.CSharp/BasicTemplateOptionStrategyAlgorithm.cs

```diff
@@ -34,34 +34,30 @@ namespace QuantConnect.Algorithm.CSharp
     /// <meta name="tag" content="filter selection" />
     public class BasicTemplateOptionStrategyAlgorithm : QCAlgorithm
     {
-        private const string UnderlyingTicker = "GOOG";
-        public readonly Symbol Underlying = QuantConnect.Symbol.Create(UnderlyingTicker, SecurityType.Equity, Market.USA);
-        public readonly Symbol OptionSymbol = QuantConnect.Symbol.Create(UnderlyingTicker, SecurityType.Option, Market.USA);
+        private Symbol _optionSymbol;
 
         public override void Initialize()
         {
             SetStartDate(2015, 12, 24);
             SetEndDate(2015, 12, 24);
             SetCash(1000000);
 
-            var equity = AddEquity(UnderlyingTicker);
-            var option = AddOption(UnderlyingTicker);
-
-            equity.SetDataNormalizationMode(DataNormalizationMode.Raw);
+            var option = AddOption("GOOG");
+            _optionSymbol = option.Symbol;
 
             // set our strike/expiry filter for this option chain
             option.SetFilter(-2, +2, TimeSpan.Zero, TimeSpan.FromDays(180));
 
             // use the underlying equity as the benchmark
-            SetBenchmark(equity.Symbol);
+            SetBenchmark("GOOG");
```

Comment:
Does this work? I half expect the `SymbolCache` to resolve `"GOOG"` to the canonical option symbol and not the equity symbol. Please confirm by stepping through to see how it gets resolved and let me know :)

Reply:
This works. Since the equity doesn't exist yet, it is created in SetBenchmark and refers to it.